pyugrid/read_netcdf.py

```diff
@@ -150,9 +150,11 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
                 raise ValueError(msg(defs['role']))
             continue
         except KeyError:
-            msg = ("File must include {} variables for {} "
-                   "named in mesh variable.").format
-            raise ValueError(msg(coord_names, defs['role']))
+            if defs['required']:
+                msg = ("File must include {} variables for {} "
+                       "named in mesh variable.").format
+                raise ValueError(msg(coord_names, defs['role']))
+            continue
```

Comment:
well, one is a missing variable ,and one a missing attribute, so we want a different error message, so this seemed like the way to to do that. Tough it looks like the message formatting has gotten lost....